ui/app/serializers/sync/destination.js

```diff
@@ -35,12 +31,14 @@ export default class SyncDestinationSerializer extends ApplicationSerializer {
     return transformedPayload;
   }
 
-  // generates id and spreads connection_details object into data
-  _normalizeFindRecord(payload) {
-    if (payload?.data?.connection_details) {
-      const { type, name, connection_details } = payload.data;
-      const id = `${type}/${name}`;
-      return { data: { id, type, name, ...connection_details } };
+  // uses name for id and spreads connection_details object into data
+  _normalizePayload(payload, requestType) {
+    if (requestType !== 'query' && payload?.data) {
+      const { data } = payload;
+      const connection_details = payload.data.connection_details || {};
+      data.id = data.name;
```

Comment:
Are you doing this instead of making the primaryKey the name?

Reply:
Yes that's right. We could set the `primaryKey` on each of the child serializers but this felt easier since we need to spread the `connection_details` anyway.